src/Patterns/templates/Scd2Synapse.twig

```diff
@@ -1 +1,95 @@
--- TODO --
+{% import "macros.twig" as macro %}
+{% set timezone = config.getTimezone() %}
+{% set inputRandomColumn = attribute(inputPrimaryKey, 0) %}
+{% set snapshotInputJoinConditionSql = macro.generateJoin(inputPrimaryKey, "snapshot", "input") %}
+{% set snapshotPrimaryKeySelect = macro.generateSnapshotPrimaryKey(snapshotPrimaryKeyParts, snapshotPrimaryKeyName) %}
+
+--- SCD2: This method tracks historical data by creating new records for new/modified data in the snapshot table --
+
+{%- if config.useDatetime() ~%}
+select
+    'datetime' as "key",
+    CONVERT(VARCHAR(255), CONVERT(DATETIMEOFFSET, CAST(CURRENT_TIMESTAMP AS DATETIME)) AT TIME ZONE '{{ timezone }}', 20) as "value"
+into "properties";
+    {% set currentDate = '(SELECT value FROM properties WHERE "key" = \'datetime\')' %}
+    {% set infiniteDate = '9999-12-31 00:00:00' | quoteValue %}
+{%- else ~%}
+select
+    'date' as "key",
+    CONVERT(VARCHAR(255), CONVERT(DATETIMEOFFSET, CAST(CURRENT_TIMESTAMP AS DATE)) AT TIME ZONE '{{ timezone }}', 20) as "value"
+into "properties";
+    {% set currentDate = '(SELECT value FROM properties WHERE "key" = \'date\')' %}
+    {% set infiniteDate = '9999-12-31' | quoteValue %}
+{%- endif -%}
+
+{%- if config.keepDeleteActive() -%}
+    {% set deletedEndDate = '9999-12-31 00:00:00' | quoteValue %}
+{%- else -%}
+    {% set deletedEndDate = currentDate %}
+{% endif %}
+
+-- Changed records: are not same in the snapshot and the input table --
+SELECT
+    {{ macro.selectFromColumns(inputColumns, "input.", true) }}
+INTO "diff_records"
+FROM {{ tableName.input | quoteIdentifier }} input
+LEFT JOIN {{ tableName.currentSnapshot | quoteIdentifier }} snapshot ON {{ macro.generateJoin(inputPrimaryKeyLower, "input", "snapshot") }}
```

Comment:
asi to nie je spravne ... ty tu robis left join iba nad hodnotami primarneho kluca ... takze ak je v input id=1, ... a v snapshot tiez id=1 ... tak sa do vysledku nedostane, ... hoci hodnoty mozu byt rozdielne ... 

Operacia `MINUS` sa inak nazyva aj `EXCEPT` a je podla vsetkaho aj v Synapse: https://docs.microsoft.com/en-us/sql/t-sql/language-elements/set-operators-except-and-intersect-transact-sql?view=sql-server-ver15

Reply:
jj tak s tím `except` to taky funguje tak je to tam

Reply:
Koukam ze jeste neco v ty scd2 upravujes tak to jeste uoravim podle toho

Reply:
ok. pozri komentare v : https://github.com/keboola/app-transformation-pattern-scd/pull/15

SNFK vyzera byt uz hotovy. Je tam odkaz na GDrive so screenshotmi z testovania.